.changeset/fluffy-panthers-hide.md

```diff
@@ -0,0 +1,5 @@
+---
+'@firebase/messaging': patch
+---
+
+Awaits onBackgroundMessage hook to avoid false-positive silent pushes warnings.
```

Comment:
`await` has a specially meaning in JS. It sounds like we are directly awaiting on the hook, but we are actually setting a timeout in parallel and await that timeout. Can you please make the changelog more specific about what we are doing and why we do it? It's probably worth mentioning the 1s limit and how we reached the number.